bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Caret.java

```diff
@@ -414,6 +414,14 @@ void setBoundsInPixels (int x, int y, int width, int height) {
 		resized = true;
 		if (isVisible && hasFocus ()) resize ();
 	}
+	setBoundsInPixels(DPIUtil.autoScaleUp(x, zoom), DPIUtil.autoScaleUp(y, zoom), sameExtent);
+}
+
+private void setBoundsInPixels (int xInPixels, int yInPixels, boolean sameExtent) {
+	boolean samePosition = this.xInPixels == xInPixels && this.yInPixels == yInPixels;
+	if (samePosition && sameExtent && isCurrentCaret()) return;
```

Comment:
Isn't this check more expensive than simply setting the (unchanged) values? Can't the values just be xet in the `setBounds` method?

Reply:
Its basically the previous implementation. The only thing that is different is that we extracted the setting of width and height from here because earlier they were set in pixels and now we set them in points, hence they are set in the setBounds method instead of setBoundsInPixels. And about the check since it was there before, I haven't bothered changing the logic. However we still need to check for extent and currentCaret as per the previous logic since they can be same even if the position is not the same. Overall this check is O(1),so I wont call it expensive.

Here, we are not setting the x and y in setBounds methods cause they are in pixels. However, another solution would be to also store x and y in points and then we can get rid of all the inPixels method. But the checks should still stay.